cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -623,47 +600,25 @@ func (m *McmManager) GetMachineDeploymentNodeTemplate(machinedeployment *Machine
 	)
 
 	switch machineClass.Kind {
-	case kindAWSMachineClass:
-		mc, err := m.awsMachineClassLister.AWSMachineClasses(m.namespace).Get(machineClass.Name)
-		if err != nil {
-			return nil, fmt.Errorf("Unable to fetch AWSMachineClass object %s, Error: %v", machineClass.Name, err)
-		}
-		awsInstance, exists := aws.InstanceTypes[mc.Spec.MachineType]
-		if !exists {
-			return nil, fmt.Errorf("Unable to fetch details for VM type %s", mc.Spec.MachineType)
-		}
-		instance = instanceType{
-			InstanceType: awsInstance.InstanceType,
-			VCPU:         awsInstance.VCPU,
-			MemoryMb:     awsInstance.MemoryMb,
-			GPU:          awsInstance.GPU,
-		}
-		region = mc.Spec.Region
-		zone = getZoneValueFromMCLabels(mc.Labels)
-	case kindAzureMachineClass:
-		mc, err := m.azureMachineClassLister.AzureMachineClasses(m.namespace).Get(machineClass.Name)
-		if err != nil {
-			return nil, fmt.Errorf("Unable to fetch AzureMachineClass object %s, Error: %v", machineClass.Name, err)
-		}
-		azureInstance, exists := azure.InstanceTypes[mc.Spec.Properties.HardwareProfile.VMSize]
-		if !exists {
-			return nil, fmt.Errorf("Unable to fetch details for VM type %s", mc.Spec.Properties.HardwareProfile.VMSize)
-		}
-		instance = instanceType{
-			InstanceType: azureInstance.InstanceType,
-			VCPU:         azureInstance.VCPU,
-			MemoryMb:     azureInstance.MemoryMb,
-			GPU:          azureInstance.GPU,
-		}
-		region = mc.Spec.Location
-		if mc.Spec.Properties.Zone != nil {
-			zone = mc.Spec.Location + "-" + strconv.Itoa(*mc.Spec.Properties.Zone)
-		}
 	case kindMachineClass:
```

Comment:
Yeah, there is one more default case which returns cloud provider not implemented error. Let's say when class kind is not MachineClass. That is why, I just retained the switch case. We can make it if-else condition if usage of switch does not seem relevant.

Reply:
the error of `not implemented` was relevant when we had seperate machineClasses , but in single machineclass I don't think its required anymore. But yes just to be safe we can keep it, and switch it to `if...else`

Reply:
We current use kind `MachineClass` but if my guess is right then we still apply provider specific MachineClass CRDs in the seed with which we can potential create such a different MachineClass.

Reply:
okay, in that case we should keep default case , but yes `if..else` would look neater.

Reply:
There is a `break` statement [here](https://github.com/gardener/autoscaler/blob/eef0d7e93cce8c45360331662b858aed0795066f/cluster-autoscaler/cloudprovider/mcm/mcm_manager.go#L619) will be violating if we introduce if else wrapper because `break` can be present only within for, switch or select blocks.

This change can be done once when the extension AWS and Azure are released by even removing the other nested switch case also that are present. Otherwise, there will be redundant codes or function calls (even if we modularise it.)

Reply:
Okay, lets leave it for a later time:thumbsup: